### PR TITLE
add functions to fasta_api with description length

### PR DESCRIPTION
**Pull request by Garonenur**

### CODE REVIEW

File: src/core/fasta_api.h [hunk 1]
```diff
@@ -29,10 +29,29 @@
 void gt_fasta_show_entry(const char *description, const char *sequence,
                          GtUword sequence_length, GtUword width, GtFile *outfp);
 
+/* Print a fasta entry with optional <description> and mandatory <sequence> to
+   <outfp>. If <width> is != 0 the sequence is formatted accordingly.
+   Will print at most <sequence_length> characters from <sequence> and at most
+   <description_length> characters from <description> if present. */
```
Comment (gordon) [inline src/core/fasta_api.h]: Why "at most" and not "exactly"?

File: src/core/fasta_api.h [hunk 2]
```diff
@@ -29,10 +29,29 @@
...
    <outstr>. If <width> is != 0 the sequence is formatted accordingly. */
 void gt_fasta_show_entry_str(const char *description, const char *sequence,
                              GtUword sequence_length, GtUword width,
                              GtStr *outstr);
 
+/* Print a fasta entry with optional <description> and mandatory <sequence> to
+   <outstr>. If <width> is != 0 the sequence is formatted accordingly.
+   Will print at most <sequence_length> characters from <sequence> and at most
+   <description_length> characters from <description> if present. */
+void gt_fasta_show_entry_nt_str(const char *description,
```
Comment (gordon) [inline src/core/fasta_api.h]: There is no `outstr` in the function prototype. Either the prototype is wrong or the function name should be different (no `_str` at the end).

File: src/core/fasta_api.h (hunk 1 above)
Comment (satta) [inline src/core/fasta_api.h]: I assume that's because if there is a '\0' in the string before the maximum length is hit, the output will stop there.

Comment (gordon) [inline src/core/fasta_api.h]: If that is the case, it should be documented accordingly. Otherwise the documentation is confusing in my opinion.

**Comment by Garonenur** [comment]
thanks for the comments, by looking closer I also realized I had missed to add one function (indirection)

**Pull request merged**